competition-simulator.md

```diff
@@ -0,0 +1,120 @@
+
+### Useful links
+
+- [Camera Controls](https://www.cyberbotics.com/doc/guide/the-3d-window#navigation-in-the-scene)
+- [Graphics settings](https://www.cyberbotics.com/doc/guide/preferences#opengl) (Useful if Webots is running slowly)
+
+## Developing your code
+
+The code for your robot is `controllers/sr_controller/sr_controller.py`. An example skeleton code is provided for convenience.
+
+<div class="warning">
+  Please refrain from editing any code in the <code>sr</code> module directory. This contains the code which maps the SR API onto the webots API.
+  Only your controller code will be present in the competition environment.
+</div>
+
+### Updates
+
+Occasionally, we may release an update to the simulation. To update, you will need to copy your code out of the `competition-simulator-master` directory, delete the directory, and re-download it using the above link. Then you can replace the new `controllers/sr_controller/sr_controller.py` with your existing code.
+
+## Robot
+
+![]({{ site.baseurl }}/resources/competition-simulator/robot-front.png)
+
+## Programming Interface
+
+**WIP**
+
+Unless otherwise stated, the simulator’s API is the same as the real SR API described in the [programming docs]({{ site.baseurl }}/programming/).
+
+### Motors
+
+Your robot has 2 motor boards attached, each with 2 motors. Board `0` has the left wheel in port `m0`, and the right wheel in `m1`. Board `1` has the gripper lift motor in `m0`, and the finger motors in `m1`.
+
+### Ruggeduino
+
+Your robot has 5 microswitches and 6 distance sensors, attached to the digital and analogue pins respectively. These are all attached to a single ruggeduino.
+
+Because these sensors are pre-attached to the ruggeduino, you do not need to set its `pin_mode`.
+
+#### Microswitches
+
+The microswitches are attached to digital pins 0-4:
+
+- Front
+- Back
+- Between gripper fingers
+- Left gripper finger
+- Right gripper finger
+
+These are shown as red coloured blocks on the robot. Using the `digital_read`  method, you'll receive a `bool` telling you whether the switch is current actuated.
+
+#### Distance Sensors
+
+Analogous to ultrasound sensors, distance sensors allow you to retrieve the distance between your robot and an object. These are attached to analogue pins 0-5:
+
+- Front Left
+- Front Right
+- Left
+- Right
+- Back Left
+- Back Right
+
+These are shown as blue coloured blocks on the robot. The `analogue_read` method will return the distance in centimetres, however only measure up to 30cm.
+
+### Camera
+
+Your robot has a camera, which is attached to the top of your gripper. A live preview of what the camera sees is shown in the top-left corner. The vision system uses Webots' object recognition, rather than [fiducial markers]({{ site.baseurl }}/programming/sr/vision/markers/).
```

Comment:
```suggestion
Your robot has a camera, which is attached to the top of your robot. A live preview of what the camera sees is shown in the top-left corner. The vision system uses Webots' object recognition, rather than [fiducial markers]({{ site.baseurl }}/programming/sr/vision/markers/).
```